index.coffee

```diff
@@ -157,29 +157,34 @@ exports.DockerProgress = class DockerProgress
 	pullProgress: (image, onProgress) ->
 		@getLayerDownloadSizes(image)
 		.then (layerSizes) ->
-			currentSize = 0
 			completedSize = 0
+			layerDownloadedSize = {}
 			totalSize = _.sum(layerSizes)
 			return (evt) ->
 				{ status } = evt
 				if status == 'Downloading'
-					currentSize = evt.progressDetail.current
+					shortId = evt.id
+					longId = _.findKey(layerSizes, (v, id) -> _.startsWith(id, shortId))
+					if longId?
```

Comment:
@Page- it was ugly to use one function for finding/checking/warning. I used a function for getting the long id and then improved error handling. please check, need to merge soon